### PR TITLE
feat(sqlsmith): add reduce query

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

1. add reduced query
2. nth_value/lead/lag second arg should be a constant
3. fix typo

- Part of #12576

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13026)

### CODE REVIEW

File: src/tests/sqlsmith/src/runner.rs [hunk 1]
```diff
@@ -164,6 +365,7 @@ impl Runner {
                     tracing::info!("query_sql: {}", query_sql);
```
Comment (b41sh) [inline src/tests/sqlsmith/src/runner.rs:365]: we can remove this log, only print the reduced sql

File: src/tests/sqlsmith/src/runner.rs [hunk 2]
```diff
@@ -128,24 +132,221 @@ impl Runner {
...
+                offset: query.offset.clone(),
+                ignore_result: false,
+            };
+            if self
+                .execute_reduce_query(conn.clone(), &reduce_query)
+                .await
+                .is_none()
+            {
+                may_fail_msg.push(format!("{:?}", query.order_by.clone()));
+                may_fail_msg.push(format!("{:?}", query.offset.clone()));
```
Comment (b41sh) [inline src/tests/sqlsmith/src/runner.rs:178]: `query.offset` can be removed

File: src/tests/sqlsmith/src/runner.rs [hunk 3]
```diff
@@ -128,24 +132,221 @@ impl Runner {
...
+                query = reduce_query.clone();
+            }
+        }
+
+        // reduce select_list
+        if let SetExpr::Select(mut select_stmt) = query.body.clone() {
+            let having = select_stmt.having.take();
+            let selection = select_stmt.selection.take();
+
+            if select_stmt.group_by.is_none() {
```
Comment (b41sh) [inline src/tests/sqlsmith/src/runner.rs:189]: don't need check `select_stmt.group_by.is_none()`

File: src/tests/sqlsmith/src/runner.rs (hunk 1 above)
Comment (TCeason) [reply src/tests/sqlsmith/src/runner.rs]: Now it is debug level.

File: src/tests/sqlsmith/src/runner.rs (hunk 2 above)
Comment (TCeason) [reply src/tests/sqlsmith/src/runner.rs]: Done

File: src/tests/sqlsmith/src/runner.rs (hunk 3 above)
Comment (TCeason) [reply src/tests/sqlsmith/src/runner.rs]: Fixed.

**Review by b41sh — APPROVED** [review]

**Pull request merged**